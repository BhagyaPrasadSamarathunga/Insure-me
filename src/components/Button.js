import React from 'react';

const Button = ({onButtonPress}) => {
    return (
        <button className={'App-button'} onClick={() => onButtonPress()}>
            Insure Me
        </button>
    );
}

export default Button;