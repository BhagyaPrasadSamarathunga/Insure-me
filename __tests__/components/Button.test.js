import React from 'react';
import TestRenderer from 'react-test-renderer';

import Button from '../../src/components/Button';

test('Button component snapshot testing', () => {
    const component = TestRenderer.create(
            <Button onButtonPress={()=>console.log("button press")}/>,
        );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); 
  });
