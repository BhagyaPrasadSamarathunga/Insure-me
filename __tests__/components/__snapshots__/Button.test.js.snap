// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot testing 1`] = `
<button
  className="App-button"
  onClick={[Function]}
>
  Insure Me
</button>
`;
